crates/viewer/re_view_graph/src/layout/provider.rs

```diff
@@ -161,24 +180,32 @@ impl ForceLayoutProvider {
         let simulation = update_simulation(simulation, params, edges, radii);
 
         Self {
-            simulation,
+            simulation: Some(simulation),
             request,
         }
     }
 
     fn layout(&self) -> Layout {
         // We make use of the fact here that the simulation is stable, i.e. the
         // order of the nodes is the same as in the `request`.
-        let mut positions = self.simulation.positions();
+        let positions: &mut dyn Iterator<Item = Pos2> = if let Some(simulation) = &self.simulation {
+            &mut simulation
+                .positions()
+                .map(|[x, y]| Pos2::new(x as f32, y as f32))
+        } else {
+            &mut self.request.all_nodes().map(|(_, v)| {
+                v.fixed_position
+                    .expect("if there is no simulation, all nodes have to be fixed")
+            })
+        };
```

Comment:
Don't erase the iterator here, this will force virtual calls for every single iteration.

Use `Either` instead so the two branches can monomorphize gently (see e.g. https://github.com/rerun-io/rerun/blob/35327062c114e8805233779043b742ae26c37260/crates/viewer/re_view/src/lib.rs#L67 for an example).

Reply:
Ah, I didn't know about `itertools::Either`. TIL :thanks: